### PR TITLE
Fix mypy typing errors in pytorch_lightning/strategies/single_tpu.py

### PR DESCRIPTION
**Pull request by Cyprien-Ricque**
## What does this PR do?

Fixes mypy typing errors in pytorch_lightning/strategies/single_tpu.py in https://github.com/Lightning-AI/lightning/issues/13445.

### Does your PR introduce any breaking changes? If yes, please list them.

This could be a breaking change in case I missed something/relative to older lightning versions.

in ```SingleTPUStrategy``` class, in function ```setup(self, trainer: "pl.Trainer")```, the call to ```self.model.on_post_move_to_device()``` has been removed because the condition to trigger it cannot be ```True```. 
The condition is ```is_overridden("on_post_move_to_device", self.lightning_module)```. To be ```True```, among other things, this condition requires the parent class to contain the ```on_post_move_to_device``` method. This ```on_post_move_to_device``` method is not present in any of the parent classes listed in ```is_overridden```, therefore the condition cannot be ```True```. (parents classes being ```LightningModule```, ```LightningDataModule``` and ```Callback```).

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/pytorch_lightning/strategies/single_tpu.py [hunk 1]
```diff
@@ -55,13 +55,10 @@ def is_distributed(self) -> bool:
         return False
 
     def setup(self, trainer: "pl.Trainer") -> None:
+        assert self.model, "self.model must be set before find_shared_parameters(self.model)"
         shared_params = find_shared_parameters(self.model)
         self.model_to_device()
-        if is_overridden("on_post_move_to_device", self.lightning_module):
-            self.model.on_post_move_to_device()
-        else:
-            set_shared_parameters(self.model, shared_params)
```
Comment (justusschock) [inline src/pytorch_lightning/strategies/single_tpu.py:63]: why did you remove this logic? Never change existing logic for the sake of typing!

File: src/pytorch_lightning/strategies/single_tpu.py [hunk 2]
```diff
@@ -70,9 +67,6 @@ def setup(self, trainer: "pl.Trainer") -> None:
         self.tpu_local_core_rank = xm.get_local_ordinal()
         self.tpu_global_core_rank = xm.get_ordinal()
 
-    def model_to_device(self) -> None:
-        self.model.to(self.root_device)
```
Comment (justusschock) [inline src/pytorch_lightning/strategies/single_tpu.py:74]: why remove this?

**Review by justusschock — CHANGES_REQUESTED** [review]

File: src/pytorch_lightning/strategies/single_tpu.py (hunk 1 above)
Comment (Cyprien-Ricque) [reply src/pytorch_lightning/strategies/single_tpu.py:63]: Sure I correct.

Put aside the fact that I don't have to do it in this PR, I changed it because as explained in the PR, It appeared to me that this ```if``` cannot be ```True```

File: src/pytorch_lightning/strategies/single_tpu.py (hunk 2 above)
Comment (Cyprien-Ricque) [reply src/pytorch_lightning/strategies/single_tpu.py:74]: Because it looked like code duplication since the function is the exact same as the one in the parent class ```SingleDeviceStrategy```. 
I noticed it because the mypy error is the same as the one in the parent class which is fixed in https://github.com/Lightning-AI/lightning/pull/13532.
Do I have to put it back?

Comment (awaelchli) [reply src/pytorch_lightning/strategies/single_tpu.py:74]: Either way is fine with me but keep in mind we would prefer focus on fixing the typing errors and not mix different types of refactors :) Otherwise LGTM

**Review by awaelchli — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Review by Borda — APPROVED** [review]
